rfcs/0000-gateway-api-metrics-exporter.md

```diff
@@ -0,0 +1,197 @@
+This will allow the new prometheus exporter to replace that project as a whole.
+The metrics will be backwards compatible with the gateway-api-state-metrics project, with the addition of new metrics.
+
+Second, implement new metrics, as per the examples below, to capture the additional status information:
+
+```promql
+gatewayapi_gateway_status_listeners_conditions{namespace="<NAMESPACE>",name="<GATEWAY>",listener_name="<LISTENER_NAME>",type="<ResolvedRefs|Ready|Other>} 1
+```
+This metric captures the status condition types of individual listeners in a Gateway.
+It will allow the status of individual named listeners to be queried, graphed and alerted on via metrics.
+The health status of listeners can then be visualised in a stat panel in Grafana, showing healthy and unhealthy listeners.
+This expands the debugging path beyond just the overall health of a Gateway.
+
+```promql
+gatewayapi_httproute_status_parents_conditions{namespace="<NAMESPACE>",name="<GATEWAY>",controller_name="<CONTROLLER_NAME>,"parent_group="<PARENT_GROUP>",parent_kind="<PARENT_KIND>",parent_name="<PARENT_NAME>",parent_namespace="<PARENT_NAMESPACE>",type="<ResolvedRefs|Accepted|Other>} 1
+```
+This metric captures the condition types for each parent of a HTTPRoute.
+The `type` field would also record any custom types set by a controller. For example, `kuadrant.io/AuthPolicyAffected` and `kuadrant.io/RateLimitPolicyAffected`.
+This will allow the health of HTTPRoutes to be reported via metrics. A HTTPRoute that has an `type` of Accepted and value of 1 means the HTTPRoute is accepted by the Gateway and can be considered healthy.
+It will also allow policy specific information about a HTTPRoute to be represented in metrics.
+For example, alerting on any HTTPRoutes that don't have the `kuadrant.io/AuthPolicyAffected` type with a value of 1 i.e. HTTPRoutes without an AuthPolicy.
+
+Tests will be added directly to the project in a similar manner to the [redis-exporter](https://github.com/oliver006/redis_exporter/blob/master/exporter/http_test.go).
+The test environment will bring up a kind cluster, create the Gateway API CRDs, example Gateway & HTTPRoute resources, then test the scrape endpoint.
+This will be the same as how [metrics are tested](https://github.com/Kuadrant/gateway-api-state-metrics/blob/main/tests/e2e.sh) for the gateway-api-state-metrics project.
+There is a [separate test function](https://github.com/Kuadrant/gateway-api-state-metrics/blob/main/tests/e2e/main_test.go) for each resource.
+
+Existing [example dashboards](https://github.com/Kuadrant/gateway-api-state-metrics/tree/main/src/dashboards) in the gateway-api-state-metrics project will be copied over to the exporter project and continue to work as before.
+However, initially it will just be the Gateway, GatewayClass and HTTPRoute dashboards as those will be the metrics that are implemented first.
+
+# Reference-level explanation
```

Comment:
So why are we not doing this work directly in gateway api state metrics. I am missing this piece . It seems we are starting with a new library?

Reply:
I see it is covered below